File: app/src/main/java/com/boardgamegeek/provider/PublishersIdCollectionProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Designers;
import com.boardgamegeek.provider.BggContract.Publishers;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class PublishersIdCollectionProvider extends BaseProvider {
	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		int publisherId = Publishers.getPublisherId(uri);
		return new SelectionBuilder().table(Tables.PUBLISHER_JOIN_GAMES_JOIN_COLLECTION).whereEquals(Publishers.PUBLISHER_ID, publisherId);
	}

	@Override
	protected String getPath() {
		return "publishers/#/collection";
	}
}

File: app/src/main/java/com/boardgamegeek/provider/PublishersIdCollectionProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import android.provider.BaseColumns._ID
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_COLLECTION
import com.boardgamegeek.provider.BggContract.Companion.PATH_PUBLISHERS
import com.boardgamegeek.provider.BggDatabase.Tables

class PublishersIdCollectionProvider : BaseProvider() {
    override val path = "$PATH_PUBLISHERS/#/$PATH_COLLECTION"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val publisherId = Publishers.getPublisherId(uri)
        return SelectionBuilder()
            .mapToTable(_ID, Tables.COLLECTION)
            .mapToTable(Games.Columns.GAME_ID, Tables.GAMES)
            .mapToTable(BggContract.Collection.Columns.UPDATED, Tables.COLLECTION)
            .mapToTable(BggContract.Collection.Columns.UPDATED_LIST, Tables.COLLECTION)
            .table(Tables.PUBLISHER_JOIN_GAMES_JOIN_COLLECTION)
            .whereEquals(Publishers.Columns.PUBLISHER_ID, publisherId)
    }
}
